src/SFA.DAS.EmployerIncentives.Commands/Persistence/ApprenticeshipIncentiveDomainRepository.cs

```diff
@@ -88,6 +91,25 @@ public async Task<List<Domain.ApprenticeshipIncentives.ApprenticeshipIncentive>>
                 select _apprenticeshipIncentiveFactory.GetExisting(incentive.Id, incentive)).ToList();
         }
 
+        public async Task<Domain.ApprenticeshipIncentives.ApprenticeshipIncentive> FindByPaymentId(Guid paymentId)
+        {
+            var payment = await _paymentDataRepository.Get(paymentId);
+
+            if (payment == null)
+            {
+                throw new ArgumentException($"Payment id {paymentId} not found");
+            }
+
+            var incentive = await _apprenticeshipIncentiveDataRepository.Get(payment.ApprenticeshipIncentiveId);
+
+            if (incentive == null)
+            {
+                throw new ArgumentException($"Apprenticeship incentive with ID of {payment.ApprenticeshipIncentiveId} and payment ID of {paymentId} not found");
```

Comment:
This should return null and the exception thrown in the command handler.  A get shouldn't throw an exception if it doesn't exist.